canonix/src/Format.hs

```diff
@@ -108,15 +131,54 @@ verbatim n = do
   src <- asksParent source
   let bs = BS.drop (startByte n) (BS.take (endByte n) src)
   write bs
+  _ <- whenSingleLineAllowed $ do
+    rw <- asksParent remainingWidth
+    tellParent mempty { singleLine = bs <$ guard (BS.length bs <= rw) }
   pure ()
 
+whenSingleLineAllowed :: CnxFmt a -> CnxFmt (Maybe a)
```

Comment:
Would be nice to have some comments for these combinations to know when to use them, it's not entirely clear to me.

Reply:
I agree it needs more documentation. We should get this thing out of the proof-of-concept sphere.

This particular function just disables some behavior when a node must be formatted in its multi-line style. It's a helper for constructing the real output functions that you actually want to use.
Needs more modules.

Reply:
Are these combinators expected to change soon? If not adding documentation (just a single line of what it does) sounds like a good time to do it.

Reply:
They'll change soon (I hope) because of #19 #18. I'll add some inline docs.